test/integration/webhook/jobs/deployment_webhook_test.go

```diff
@@ -22,111 +22,124 @@ import (
-						)
-				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+		ginkgo.It("should allow to change the queue", func() {
+			createdDeployment := &appsv1.Deployment{}
 
-				ginkgo.By("Updating the deployment should not allow to change the queue name", func() {
-					deploymentToUpdate := &appsv1.Deployment{}
-					gomega.Expect(k8sClient.Get(ctx, lookupKey, deploymentToUpdate)).Should(gomega.Succeed())
-					deploymentWrapper := &testingdeployment.DeploymentWrapper{
-						Deployment: *deploymentToUpdate,
-					}
+			ginkgo.By("Change queue name", func() {
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(deployment), createdDeployment)).Should(gomega.Succeed())
+					deploymentWrapper := &testingdeployment.DeploymentWrapper{Deployment: *createdDeployment}
 					updatedDeployment := deploymentWrapper.Queue("another-queue").Obj()
-					gomega.Expect(k8sClient.Update(ctx, updatedDeployment)).To(gomega.HaveOccurred())
-				})
+					g.Expect(k8sClient.Update(ctx, updatedDeployment)).To(testing.BeForbiddenError())
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
 			})
 		})
 
-		ginkgo.When("The queue-name label is not set", func() {
-			var (
-				deployment *appsv1.Deployment
-				lookupKey  types.NamespacedName
-			)
+		ginkgo.It("should not allow to remove the queue label", func() {
+			createdDeployment := &appsv1.Deployment{}
 
-			ginkgo.BeforeEach(func() {
-				deployment = testingdeployment.MakeDeployment("deployment-without-queue-name", ns.Name).
-					Obj()
-				lookupKey = client.ObjectKeyFromObject(deployment)
+			ginkgo.By("Remove queue label", func() {
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(deployment), createdDeployment)).Should(gomega.Succeed())
+				delete(createdDeployment.Labels, constants.QueueLabel)
+				gomega.Expect(k8sClient.Update(ctx, createdDeployment)).To(testing.BeForbiddenError())
 			})
+		})
+	})
 
-			ginkgo.It("Should not inject queue name to pod template labels", func() {
+	ginkgo.When("the queue-name label is not set", func() {
+		ginkgo.BeforeEach(func() {
+			ginkgo.By("Create deployment", func() {
+				deployment = testingdeployment.MakeDeployment("deployment", ns.Name).Obj()
 				gomega.Expect(k8sClient.Create(ctx, deployment)).Should(gomega.Succeed())
+			})
+		})
+
+		ginkgo.It("should not inject queue name to pod template labels", func() {
+			createdDeployment := &appsv1.Deployment{}
+
+			gomega.Eventually(func(g gomega.Gomega) {
+				g.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(deployment), createdDeployment)).Should(gomega.Succeed())
+				g.Expect(createdDeployment.Spec.Template.Labels[constants.QueueLabel]).
+					To(
+						gomega.BeEmpty(),
+						"Queue name should not be injected to pod template labels",
+					)
+			}, util.Timeout, util.Interval).Should(gomega.Succeed())
+		})
+
+		ginkgo.It("should not allow to change the queue name", func() {
```

Comment:
should not allow to introduce the queue name, as it was not existent

Reply:
No, this test means that we can't change queue-name at all.

Reply:
Ah, I see. Make sense.

Reply:
Done